R/run.R

```diff
@@ -339,12 +343,28 @@ run_bm <- function(bm, ..., n_iter = 1, batch_id = NULL, profiling = FALSE,
   })
 
   results <- list()
+  error <- NULL
   for (i in seq_len(n_iter)) {
-    results[[i]] <- run_iteration(
-      bm = bm,
-      ctx = ctx,
-      profiling = profiling,
-      drop_caches = global_params[["drop_caches"]]
+    tryCatch(
+      {
+        results[[i]] <- run_iteration(
+          bm = bm,
+          ctx = ctx,
+          profiling = profiling,
+          drop_caches = global_params[["drop_caches"]]
+        )
+      },
+      error = function(e){
+        # Note: this will only capture the last error
+        error <<- list(
+          error = as.character(e),
+          stack_trace = vapply(
+            traceback(3),
+            function(x) paste(x, collapse = "\n"),
+            character(1)
+          )
+        )
+      }
```

Comment:
Previously we were handling errors higher up, but here seems like the right place—if an iteration errors out, it should get caught, recorded, and escalated.

Reply:
Also open to putting/naming what goes in `error` differently; `result.error.error` is literally the string of the error, which is hard to see from the traceback. Maybe we should switch to `rlang::trace_back()` or something?